CRM/Core/BAO/Managed.php

```diff
@@ -54,4 +79,12 @@ public static function isApi4ManagedType(string $entityName) {
     return $type && in_array('ManagedEntity', $type, TRUE);
   }
 
+  /**
+   * Options callback for `base_module`.
+   * @return array
+   */
+  public static function getBaseModules(): array {
+    return array_merge(['civicrm' => ts('CiviCRM Core')], CRM_Core_PseudoConstant::getExtensions());
```

Comment:
@colemanw -- Ah, let's see if I follow. When `civicrm_managed` was exposed to APIv4, it needed a method to lookup legal values of `module`. `getExtensions()`  was used. But this use-case shows that it's not quite accurate enough for this purpose. So this patch adds a method which should be more accurate.

There is a pre-existing method that is *even more* accurate -- `CRM_Core_Module::getAll()`. That's the method used by  [`CRM_Core_ManagedEntities`](https://github.com/civicrm/civicrm-core/blob/6658626809ee1837cced2413509996fb23d2c13a/CRM/Core/ManagedEntities.php#L39) and corresponds to some other options in [`hook_managed` docs](https://docs.civicrm.org/dev/en/latest/hooks/hook_civicrm_managed/) (e.g. `drupal.mymodule`). Specifically, `Module::getAll()` provides:

* `civicrm`
* CiviCRM module-extensions (full `key`)
* Drupal/Backdrop modules (with `drupal.*` prefix)
* Joomla plugin-extensions (with `joomla.*` prefix)

(The essence of a "managed" entity is to install/delete based on whether the `module` is active/inactive, which requires inspecting `module` statuses. `CRM_Core_Module`  is the list of inspectable modules.)

Reply:
@totten `CRM_Core_Module::getAll()` doesn't return human-readable labels.

Reply:
@totten the user-facing label is shown for example in the SK UI and in the Afform UI, both in the "Package" column and in the "Filter by Package" dropdown:
![image](https://github.com/civicrm/civicrm-core/assets/2874912/0a316d26-8505-462d-b5b3-cd0182e7ef6e)

Reply:
Aah, cool. I see it now. :+1: